### ISSUE TITLE
Exception while importing playlists

### ISSUE THREAD

**Issue by orontee** [suaviloquence/mopidy-listenbrainz#7]
# Context

Current version triggers an exception while importing playlists.

```
juin 18 17:53:27 argos mopidy[21273]: INFO     [MainThread] mopidy.commands Starting GLib mainloop
juin 18 17:53:28 argos mopidy[21273]: DEBUG    [ListenbrainzFrontend-15 (_actor_loop)] mopidy_listenbrainz.frontend Listenbrainz token valid!
juin 18 17:53:28 argos mopidy[21273]: DEBUG    [ListenbrainzFrontend-15 (_actor_loop)] mopidy_listenbrainz.frontend Will limit track searches to URIs: ('local:',)
juin 18 17:53:28 argos mopidy[21273]: INFO     [ListenbrainzFrontend-15 (_actor_loop)] mopidy_listenbrainz.frontend Importing ListenBrainz playlists
juin 18 17:53:28 argos mopidy[21273]: ERROR    [ListenbrainzFrontend-15 (_actor_loop)] pykka Unhandled exception in ListenbrainzFrontend (urn:uuid:4d386f3c-0a12-4582-9254-76d55dad0996):

juin 18 17:53:28 argos mopidy[21273]: Traceback (most recent call last):
juin 18 17:53:28 argos mopidy[21273]:   File "/usr/local/lib/python3.11/dist-packages/pykka/_actor.py", line 230, in _actor_loop_setup
juin 18 17:53:28 argos mopidy[21273]:     self.on_start()
juin 18 17:53:28 argos mopidy[21273]:   File "/usr/local/lib/python3.11/dist-packages/mopidy_listenbrainz/frontend.py", line 50, in on_start
juin 18 17:53:28 argos mopidy[21273]:     self.import_playlists()
juin 18 17:53:28 argos mopidy[21273]:   File "/usr/local/lib/python3.11/dist-packages/mopidy_listenbrainz/frontend.py", line 60, in import_playlists
juin 18 17:53:28 argos mopidy[21273]:     playlist_datas = self.lb.list_playlists_created_for_user()
juin 18 17:53:28 argos mopidy[21273]:                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
juin 18 17:53:28 argos mopidy[21273]:   File "/usr/local/lib/python3.11/dist-packages/mopidy_listenbrainz/listenbrainz.py", line 179, in list_playlists_created_for_user
juin 18 17:53:28 argos mopidy[21273]:     playlist_data = self._collect_playlist_data(playlist_identifier)
juin 18 17:53:28 argos mopidy[21273]:                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
juin 18 17:53:28 argos mopidy[21273]:   File "/usr/local/lib/python3.11/dist-packages/mopidy_listenbrainz/listenbrainz.py", line 236, in _collect_playlist_data
juin 18 17:53:28 argos mopidy[21273]:     track_mbid = track_identifier_to_mbid(track_identifier)
juin 18 17:53:28 argos mopidy[21273]:                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
juin 18 17:53:28 argos mopidy[21273]:   File "/usr/local/lib/python3.11/dist-packages/mopidy_listenbrainz/listenbrainz.py", line 46, in track_identifier_to_mbid
juin 18 17:53:28 argos mopidy[21273]:     track_path = urlparse(track_identifier).path
juin 18 17:53:28 argos mopidy[21273]:                  ^^^^^^^^^^^^^^^^^^^^^^^^^^
juin 18 17:53:28 argos mopidy[21273]:   File "/usr/lib/python3.11/urllib/parse.py", line 385, in urlparse
juin 18 17:53:28 argos mopidy[21273]:     url, scheme, _coerce_result = _coerce_args(url, scheme)
juin 18 17:53:28 argos mopidy[21273]:                                   ^^^^^^^^^^^^^^^^^^^^^^^^^
juin 18 17:53:28 argos mopidy[21273]:   File "/usr/lib/python3.11/urllib/parse.py", line 124, in _coerce_args
juin 18 17:53:28 argos mopidy[21273]:     return _decode_args(args) + (_encode_result,)
juin 18 17:53:28 argos mopidy[21273]:            ^^^^^^^^^^^^^^^^^^
juin 18 17:53:28 argos mopidy[21273]:   File "/usr/lib/python3.11/urllib/parse.py", line 108, in _decode_args
juin 18 17:53:28 argos mopidy[21273]:     return tuple(x.decode(encoding, errors) if x else '' for x in args)
juin 18 17:53:28 argos mopidy[21273]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
juin 18 17:53:28 argos mopidy[21273]:   File "/usr/lib/python3.11/urllib/parse.py", line 108, in <genexpr>
juin 18 17:53:28 argos mopidy[21273]:     return tuple(x.decode(encoding, errors) if x else '' for x in args)
juin 18 17:53:28 argos mopidy[21273]:                  ^^^^^^^^
juin 18 17:53:28 argos mopidy[21273]: AttributeError: 'list' object has no attribute 'decode'
```

# Analysis

It looks like the `identifier` attribute of track DTOs can be a list of string (note that the [API documentation](https://listenbrainz.readthedocs.io/en/latest/users/api/playlist.html#get--1-user-(playlist_user_name)-playlists) isn't that precise on the response content...

I don't know whether it's a change in the API or my listening habits that have evolved :smiley:  But the six recommendation playlists I have, all trigger the exception.

A patch will be made available in very short time.

**Comment by orontee** [comment]
@suaviloquence I confirm the proposed commit fix the problem.

**Comment by suaviloquence** [comment]
fixed by #8
